Assignment-5/CarDrive.ts

```diff
@@ -0,0 +1,131 @@
+import {
+    CarAbrupltyStopException,
+    CollisionAccidentException,
+    HitAndRunAccidentException,
+    EnvironmentalAccidentException,
+    CarBrakeFailException
+  } from './exceptions';
+    
+  try {
+    driveCar();
+  } catch (e) {
+    if (e instanceof CarBrakeFailException) {
+      console.log("Car Brake Fail Exception occurred.");
+      handleBreakFailDamage();
+    } else if (e instanceof CollisionAccidentException) {
+      console.log("Collision Accident Exception occurred.");
+      handleCollisionAccident();
+    } else if (e instanceof EnvironmentalAccidentException) {
+      console.log("Environmental Accident Exception occurred.");
+      handleNatureRelatedAccident();
+    } else if (e instanceof HitAndRunAccidentException) {
+      console.log("Hit and Run Accident Exception occurred.");
+      handleHitAndRunAccident();
+    }else if (e instanceof CarAbrupltyStopException){
+      console.log("Car Abruplty Stop Exception occurred.");
+      AbrupltyException();
+    } else {
+      console.error("An error occurred:", e.message);
+    }
+  } finally {
+    cleanUpAfterAccident();
+  }
+
+  function driveCar() {
+    const isCarBreakFail = true;
+    const isCollision = false;
+    const isEnvironmental = false;
+    const isAbrupltyStop = false;
+  
+    if (isCarBreakFail) {
+      throw new CarBrakeFailException(
+        "Car Break Fail due to NonService of car, Have proper service"
+      );
+    } else if (isCollision) {
+      throw new CollisionAccidentException(
+        "Collision with another vehicle occurred."
+      );
+    } else if (isAbrupltyStop) {
+      throw new CarAbrupltyStopException (
+        "Automatically stop and not starting."
+      );
+    } else if (isEnvironmental) {
+      throw new EnvironmentalAccidentException(
+        "Environmental conditions such as thunderstorm. Drive with caution."
+      );
+    } else {
+      throw new HitAndRunAccidentException("Hit and run incident happened.");
+    }
+  }
+  
+  function handleBreakFailDamage() {
```

Comment:
all these functions below are help in handling the exceptions, so I think we should either move it in Exceptions.ts or rather in a separate file.